app/controllers/orgs_controller.rb

```diff
@@ -17,26 +18,13 @@ def admin_update
     @org = Org.find(params[:id])
     authorize @org
     @org.logo = attrs[:logo] if attrs[:logo]
-    failure = ''
     tab = (attrs[:feedback_enabled].present? ? 'feedback' : 'profile')
-    
-    if attrs[:links].present?
-      if is_json_array_of_objects?(attrs[:links])
-        json = JSON.parse(attrs[:links])
-        # Make sure that the JSON hash is structured as: {"link":"string","text":"string"}
-        if json.all?{ |o| o['link'].present? && o['text'].present? }
-          @org.links = json
-        else
-          failure = _('Unable to save your changes. Invalid URLs.')
-        end
-      else
-        failure = _('Unable to save your changes. Invalid URLs.')
-      end
-      attrs.delete('links')
+    if params[:org_links].present?
+      @org.links = JSON.parse(params[:org_links]) 
```

Comment:
nice refactor! We might need to to rescue if any parse error is arisen. Also in templates_controller.rb